pyang/translators/yang.py

```diff
@@ -110,18 +163,23 @@ def emit_stmt(ctx, stmt, fd, level, prev_kwd_class, indent, indentstep):
     kwd_class = get_kwd_class(stmt.keyword)
     if ((level == 1 and
          kwd_class != prev_kwd_class and kwd_class != 'extension') or
-        stmt.keyword in _keyword_with_trailing_newline):
+        (level == 1 and stmt.keyword in
+         _keyword_with_trailing_newline_toplevel(ctx)) or
+        stmt.keyword in _keyword_with_trailing_newline(ctx)):
         fd.write('\n')
 
     if stmt.keyword == '_comment':
         emit_comment(stmt.arg, fd, indent)
         return
 
     fd.write(indent + keywordstr)
-    if stmt.arg != None:
-        if (stmt.keyword in _keyword_prefer_squote_arg and
-            stmt.arg.find("'") == -1):
-            fd.write(" '" + stmt.arg + "'")
+    if stmt.arg is not None:
+        forcenlarg = _force_newline_arg(ctx)
+        if '\n' in stmt.arg:
+            emit_arg(stmt, fd, indent, indentstep, forcenlarg)
+        elif stmt.keyword in _keyword_prefer_squote_arg(ctx):
+            emit_arg_squote(stmt.keyword, stmt.arg, fd, indent,
+                            ctx.max_line_len)
```

Comment:
Note this use of the `--max-line-len` option value. Is this OK here (because YANG output is perhaps a bit special) or would it be better to define a separate `--yang-max-line-len` option (c.f. `--tree-max-line-len`)?

Reply:
OK.